atc/creds/credhub/credhub.go

```diff
@@ -58,16 +58,16 @@ func (c CredHubAtc) findCred(path string) (credentials.Credential, bool, error)
 		return cred, false, err
 	}
 
-	_, err = ch.FindByPath(path)
+	_, err = ch.FindByPartialName(path)
```

Comment:
Are these able to be used interchangeably? Wouldn't the path be different than the name in credhub? For example, path would be `/credhub/concourse/username` versus name would be `username`?

Reply:
It looks like the `FindByPath` is more restrictive than `FindByPartialName`? 🤔  I'm surprised that `FindByPath` did not find anything for `/concourse/main/team_secret` when there is a var in that exact path?

Was there anything that changed in the `FindByPath` method in credhub in this new bump that is making us need to use `FindByPartialName` instead now?

Reply:
> I'm surprised that `FindByPath` did not find anything for `/concourse/main/team_secret` when there is a var in that exact path?

the reason is credhub take "path" as a strictly meaning a path to the cred i.e. the parent folder of where the cred is, but not the complete uri (in Concourse semantics) to the cred.

thats also the main reason I made the switch of the method. 
The second reason is `FindByPartialName` failed earlier so we could skip the next api call if not found. 
Third reason is the linked commit from Credhub CLI actually made the error comparison harder.  `FindByParthialName` is at least returning and EXPOSING an error `NewNoMatchingCredentialsFoundError` that we can use. If not, we will have to do an awkward error comparison by comparing the error message, which would be broken if next time they change the error message.



> Was there anything that changed in the `FindByPath` method in credhub in this new bump that is making us need to use `FindByPartialName` instead now?

No, the behaviour of both method are the same. Its just us were using the `FindByPath` differently.

Reply:
Hmm I see, thanks for explaining everything!! :)